File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/structure/foreignkey/ForeignModelNoReferences.java
package com.raizlabs.android.dbflow.test.structure.foreignkey;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ForeignKey;
import com.raizlabs.android.dbflow.annotation.ModelContainer;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.test.TestDatabase;
import com.raizlabs.android.dbflow.test.structure.TestModel1;

/**
 * Description:
 */
@ModelContainer
@Table(database = TestDatabase.class)
public class ForeignModelNoReferences extends TestModel1 {

    @Column
    @ForeignKey
    ForeignParentModel parentModel;

    @Column
    @ForeignKey(tableClass = ForeignParentModel.class)
    String parentName;

    @Column
    @ForeignKey
    ForeignParentModel2 parentModel2;
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/structure/foreignkey/ForeignModelNoReferences.kt
package com.raizlabs.android.dbflow.test.structure.foreignkey

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.ForeignKey
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.test.TestDatabase
import com.raizlabs.android.dbflow.test.structure.TestModel1

/**
 * Description:
 */
@Table(database = TestDatabase::class)
class ForeignModelNoReferences : TestModel1() {

    @Column
    @ForeignKey
    var parentModel: ForeignParentModel? = null

    @Column
    @ForeignKey(tableClass = ForeignParentModel::class)
    var parentName: String? = null

    @Column
    @ForeignKey
    var parentModel2: ForeignParentModel2? = null
}
